autogluon/utils/tabular/data/utils.py

```diff
@@ -0,0 +1,34 @@
+import logging
+import math
+
+from pandas import DataFrame
+
+from ..features.types import get_type_map_raw
+from ..features.feature_metadata import R_INT, R_FLOAT, R_CATEGORY
+
+
+logger = logging.getLogger(__name__)
+
+
+# TODO: Documentation
+def get_approximate_df_mem_usage(df: DataFrame, sample_ratio=0.2):
```

Comment:
0.2 is safe, I'm mainly trying to avoid a crash which could occur if the DF was ~50%, since 0.2 is only 10%. If the DF is 80%, we would get various crashes elsewhere before this function caused a crash.